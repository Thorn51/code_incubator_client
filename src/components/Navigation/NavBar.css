.navigationBar {
  padding: 0 10px 0 10px;
  background-color: #01162790;
  color: white;
}

.navigation_list,
.nav_logo {
  list-style-type: none;
}

.nav_logo {
  float: left;
}

.nav_li {
  float: right;
  font-size: 1em;
}

li a {
  display: block;
  padding: 8px;
  color: white;
  text-decoration: none;
}
nav {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 10;
}
