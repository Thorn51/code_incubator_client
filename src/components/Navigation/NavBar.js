import React from "react";
import { Link } from "react-router-dom";
import "./NavBar.css";
import TokenServices from "../../services/token-service";

class NavBar extends React.Component {
  handleLogoutClick = () => {
    TokenServices.clearAuthToken();
  };

  //navigation if user is logged in
  renderLogoutNav() {
    return (
      <nav className="navigationBar">
        <ul className="nav_logo">
          <li className="nav_li">
            <Link to="/">Code Incubator</Link>
          </li>
        </ul>
        <ul className="navigation_list">
          <li className="nav_li">
            <Link onClick={this.handleLogoutClick} to="/">
              Logout
            </Link>
          </li>
          <li className="nav_li">
            <Link to="/submitidea">Submit Idea</Link>
          </li>
        </ul>
      </nav>
    );
  }

  //Navigation if user is not logged in
  renderLoginNav() {
    return (
      <nav className="navigationBar">
        <ul className="nav_logo">
          <li className="nav_li">
            <Link to="/">Code Incubator</Link>
          </li>
        </ul>
        <ul className="navigation_list">
          <li className="nav_li">
            <Link to="/login">Login</Link>
          </li>
          <li className="nav_li">
            <Link to="/register">Register</Link>
          </li>
        </ul>
      </nav>
    );
  }
  render() {
    return (
      <>
        {/* Render navigation based on if the user is logged in or not */}
        {TokenServices.hasAuthToken()
          ? this.renderLogoutNav()
          : this.renderLoginNav()}
      </>
    );
  }
}

export default NavBar;
