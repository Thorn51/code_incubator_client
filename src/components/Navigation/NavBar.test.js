import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import NavBar from "./NavBar";

describe("<NavBar />", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("Renders as expected", () => {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
