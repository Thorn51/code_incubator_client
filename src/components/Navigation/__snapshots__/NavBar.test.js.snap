// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> Renders as expected 1`] = `
<nav
  className="navigationBar"
>
  <ul
    className="nav_logo"
  >
    <li
      className="nav_li"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Code Incubator
      </a>
    </li>
  </ul>
  <ul
    className="navigation_list"
  >
    <li
      className="nav_li"
    >
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
    <li
      className="nav_li"
    >
      <a
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
  </ul>
</nav>
`;
